pydmd/havok.py

```diff
@@ -0,0 +1,194 @@
+        # are given by self.dmd_timesteps, and the initial state values are
+        # given by the initial values of the linear time-delay embeddings.
+        # This yields a reconstruction of V from the SVD of the hankel matrix.
+        reconstructed_v = signal.lsim(
+            havok_system,
+            U=self.forcing_input,
+            T=self.dmd_timesteps[:len(self.forcing_input)],
+            X0=self.linear_embeddings[0, :]
+        )[1]
+
+        # Compute a reconstruction of the original data x by first recomputing
+        # the hankel matrix with the reconstructed V matrix and then recovering
+        # a 1-D time-series from the computed hankel matrix.
+        reconstructed_hankel_matrix = np.linalg.multi_dot([
+            self._svd_modes[:, :-1],
+            np.diag(self._svd_amps[:-1]),
+            reconstructed_v.conj().T
+        ])
+        reconstructed_x = self._dehankel(reconstructed_hankel_matrix)
+
+        return reconstructed_x
+
+    def fit(self, x, dt):
+        """
+        Perform HAVOK analysis on 1-D time-series data x given the size of
+        the time step dt separating the observations in x.
+        """
+        self._snapshots, self._snapshots_shape = self._col_major_2darray(x)
+        self._snapshots = self._snapshots.squeeze()
+
+        # Check that input data is a 1D time-series
+        if np.ndim(self._snapshots) != 1:
+            raise ValueError("Input data must be a 1-D time series.")
+
+        # Get number of data points
+        n_samples = len(self._snapshots)
+
+        # Check that the input time-series contains enough observations
+        if n_samples < self._d:
+            msg = """The number of snapshots provided is not enough for d={}.
+Expected at least d."""
+            raise ValueError(msg.format(self._d))
+
+        # Compute hankel matrix for the input data
+        self._hankel_matrix = self._pseudo_hankel_matrix(
+            self._snapshots.reshape(1, -1)
+        )
+
+        # Take SVD of the hankel matrix
+        # Save the resulting U, s, and V for future reconstructions
+        self._svd_modes, self._svd_amps, self._embeddings = compute_svd(
+            self._hankel_matrix, self.svd_rank
+        )
+        self._r = len(self._svd_amps)
+
+        # Perform DMD on the time-delay embeddings
+        self._sub_dmd.fit(self._embeddings.T)
+
+        # Compute the full (discrete-time) system matrix
+        w, e = self._sub_dmd.modes, self._sub_dmd.eigs
+        regression_model_discrete = np.linalg.multi_dot(
+            [w, np.diag(e), np.linalg.pinv(w)]
+        )
+
+        # Roughly approximate the continuous-time system matrix
+        regression_model_continuous = (
+            regression_model_discrete - np.eye(self._r)
+        ) / dt
+
+        # Save A matrix and B vector 
+        self._A = regression_model_continuous[:-1, :-1]
+        self._B = regression_model_continuous[:-1, -1].reshape(-1, 1)
```

Comment:
It's the same but it's more readable:
```suggestion
        self._B = regression_model_continuous[:-1, -1, None]
```